components/brave_extension/extension/brave_extension/content_element_picker.ts

```diff
@@ -0,0 +1,424 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Notify the background script as soon as the content script has loaded.
+// chrome.tabs.insertCSS may sometimes fail to inject CSS in a newly navigated
+// page when using the chrome.webNavigation API.
+// See: https://bugs.chromium.org/p/chromium/issues/detail?id=331654#c15
+// The RenderView should always be ready when the content script begins, so
+// this message is used to trigger CSS insertion instead.
+
+let pickerFrame: HTMLIFrameElement | null
+
+// When the picker is activated, it eats all pointer events and takes up the
+// entire screen. All calls to document.elementFromPoint(..) will return the
+// frame. We disable pointer events for the duration of the query to get
+// around this.
+const elementFromFrameCoords = (x: number, y: number): Element | null => {
+  if (pickerFrame === null) { return null }
+  pickerFrame.style.setProperty('pointer-events', 'none', 'important')
+  const elem = document.elementFromPoint(x, y)
+  pickerFrame.style.setProperty('pointer-events', 'auto', 'important')
+  return elem
+}
+
+enum SpecificityFlags {
```

Comment:
nit: If this is Brave authored code, you might consider doing something other than Bitwise operations, since these are all exclusive cases (so you dont have the bitwise benefit of a single value being able to encode mutual, non contradictory cases), and you're not going to see a perf benefit in most (all?) JS engines that aren't running in ASM or similar mode IIRC

Reply:
Understood (and again this is just questions, not requests for change) but i mean that since
1. all the cases in SpecificityFlags are mutually exclusive (no mast will ever match more than one value) and
2. in JS enums are doubles, so bitwise operators require casting to int32 (its possible that v8 is able to optimize the case since the latent type is stable)

Its both faster, and usually clearer to just check with `===` (which won't require conversion). 

Anyway, take it with a grain of salt, I haven't been paid (primarily) to write good JS in a decade, but just adding in case its helpful :)

Reply:
discussed in DM